addons/hr_work_entry_contract/models/hr_contract.py

```diff
@@ -100,6 +101,35 @@ def _get_attendance_intervals(self, start_dt, end_dt):
             ))
         return result
 
+    def _postprocess_attendance_intervals(self, intervals):
+        # _attendance_intervals_batch combines the attendances regardless of work entry type if their date and time overlap.
+        # This makes a single attendance block with the work entry type of the first attendance only. This function undoes those
+        # undesirable merges
+        # intervals: [(start_date, end_date, attendances)]
+        try:
+            multi_type_intervals = [interval for interval in intervals if len(interval[2].mapped('work_entry_type_id')) > 1]
+        except KeyError:
+            return intervals
+        for interval in multi_type_intervals:
+            attendances = interval[2]
+            current_work_entry_type = attendances[0].work_entry_type_id
+            attendances_of_type = self.env["resource.calendar.attendance"]
+            start_date = interval[0]
+            for attendance in attendances:
+                if attendance.work_entry_type_id == current_work_entry_type:
+                    end_date = interval[0].replace(hour=floor(attendance.hour_to), minute=int((attendance.hour_to % 1) * 60))
+                    attendances_of_type += attendance
+                else:
+                    intervals._items.append((start_date, end_date, attendances_of_type))
+                    start_date = interval[0].replace(hour=floor(attendance.hour_from), minute=int((attendance.hour_from % 1) * 60))
+                    end_date = interval[0].replace(hour=floor(attendance.hour_to), minute=int((attendance.hour_to % 1) * 60))
+                    current_work_entry_type = attendance.work_entry_type_id
+                    attendances_of_type += attendance
```

Comment:
> I feel like this might not work if you have a situation with wet_1/wet_2/wet_1 which will probably create overlapping results.

As far as i tested, it would create 3 blocks but that don't overlap (each block will have the start/end of the original attendance).

> Additionally, the situation wet_1/wet_2/wet_2 will create three blocks instead of 2 if I'm not mistaken.

As far as I can tell, we will only have 2 block. The loop just accumulate until the work type change.

Reply:
> I feel like this might not work if you have a situation with wet_1/wet_2/wet_1 which will probably create overlapping results

wet_1/wet_2/wet_1 seems fine to me as well, attendances cannot overlap beyond having the same bounds so I don't see a case where this would go differently.

> Additionally, the situation wet_1/wet_2/wet_2 will create three blocks instead of 2 if I'm not mistaken.

I only get 2 blocks when I try.

> Last point is that even though I don't feel it'll be of use here, it's always better to use |= rather than += when working with recordsets, as + might create duplicates

I didn't think of that, nice catch !

Reply:
You're both right, I misread the way the method was done. My bad!